src/trainutils.jl

```diff
@@ -119,110 +116,36 @@ function bubbleKernel(x, r::Float64)
     return bubbleKernelSqScalar.(x.^2,r^2)
 end
 
-
 """
-    distMatrix(grid::Array, toroidal::Bool)::Array{Float64, 2}
-
-Return the distance matrix for a non-toroidal or toroidal SOM.
+    thresholdKernel(x, r::Float64)
 
-# Arguments
-- `grid`: coordinates of all neurons as generated by one of the `grid-`functions
-with x-coordinates in 1st column and y-coordinates in 2nd column.
-- `toroidal`: true for a toroidal SOM.
+Simple FlowSOM-like hard-threshold kernel
 """
-function distMatrix(grid::Array, toroidal::Bool)::Array{Float64, 2}
-
-    X = 1
-    Y = 2
-    xdim = maximum(grid[:,X]) - minimum(grid[:,X]) + 1.0
-    ydim = maximum(grid[:,Y]) - minimum(grid[:,Y]) + 1.0
-
-    numNeurons = size(grid,1)
-
-    dm = zeros(Float64, (numNeurons,numNeurons))
-    for i in 1:numNeurons
-        for j in 1:numNeurons
-            Δx = abs(grid[i,X] - grid[j,X])
-            Δy = abs(grid[i,Y] - grid[j,Y])
-
-            if toroidal
-                Δx = min(Δx, xdim-Δx)
-                Δy = min(Δy, ydim-Δy)
-            end
-
-            dm[i,j] = √(Δx^2 + Δy^2)
-        end
+function thresholdKernel(x, r::Float64, maxRatio=4/5, zero=1e-6)
+    if r >= maxRatio*maximum(x) #prevent smoothing everything to a single point
+        r = maxRatio*maximum(x)
     end
-
-    return dm
+    return zero.+(x.<=r)
 end
 
-
 """
-    normTrainData(train::Array{Float64, 2}, norm::Symbol)
+    distMatrix(metric=Chebyshev())
 
-Normalise every column of training data.
+Return a function that uses the `metric` (compatible with metrics from package `Distances`) calculates distance matrixes from normal row-wise data matrices, using the `metric`.
 
-# Arguments
-- `train`: DataFrame with training Data
-- `norm`: type of normalisation; one of `minmax, zscore, none`
+Use as a parameter of `trainGigaSOM`.
 """
-function normTrainData(train::Array{Float64, 2}, norm::Symbol)
+function distMatrix(metric=Chebyshev())
+    return (grid::Matrix{Float64}) -> begin
```

Comment:
why do you pack everything in a return statement? would be cleaner to simply do all operations and then return.